### PR TITLE
[feat ] Team & Team Member Pages

### PR DESCRIPTION
**Pull request by hstrowd**
Adds a page for the entire hatch team and individual pages for each team member, including a list of the blog posts they have published. Refactors the gatsby-node.js file to decompose it into a handful of smaller, more manageable modules.

Basecamp To-Do: https://3.basecamp.com/4029961/buckets/7988885/todos/1611141773

@hatchloyalty/engineering

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -4,214 +4,12 @@
...
-
-          resolve()
-        })
-      })
-    ))
-
-    Promise.all(tagPagesCreation).then(resolve)
-  })
-})
+const { createPages: createBlogPages } = require('./config/blogs-setup')
```
Comment (DLavin23) [inline gatsby-node.js]: That's a lot cleaner!

File: src/components/layout.tsx [hunk 2]
```diff
@@ -23,9 +23,8 @@ const Main = system({
   is: 'main',
 })
 
-export interface IPureLayoutProps {
+export interface IPureLayoutProps extends ILayoutProps {
```
Comment (DLavin23) [inline src/components/layout.tsx]: Just curious, why did you rename from `IPureLayoutProps` to `ILayoutProps`?

File: content/team-members/dan-lavin.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "active": true,
```
Comment (DLavin23) [inline content/team-members/dan-lavin.json]: /nit, but you are technically missing Camille ;)

File: src/components/seo.tsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ type MetaEntry =
   | { name: string; content: any; property?: undefined }
   | { property: string; content: any; name?: undefined }
 
-export interface IPureSEOProps {
+export interface IPureSEOProps extends ISEOProps {
```
Comment (DLavin23) [inline src/components/seo.tsx]: Same question, why did you rename from `IPureSEOProps` to `ISEOProps`?

File: src/templates/team-member.tsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        }
+      }
+    }
+    allTeamMembersJson(
+      filter: { name: { eq: $name } }
+    ) {
+      edges {
+        node {
+          name
+          position
```
Comment (DLavin23) [inline src/templates/team-member.tsx]: So, at a high level, if I wanted to add a `bio` or `description`, I would need to add it to the team member JSON file and then update this query. Am I understanding that right?

File: src/components/seo.tsx (hunk 4 above)
Comment (hstrowd) [reply src/components/seo.tsx]: Again this allowed me to have all prop definitions flow through from `ISEOProps` into `IPureSEOProps`.

File: src/templates/team-member.tsx (hunk 5 above)
Comment (hstrowd) [reply src/templates/team-member.tsx]: That's correct.

File: src/components/layout.tsx (hunk 2 above)
Comment (DLavin23) [reply src/components/layout.tsx]: gotcha, makes sense.
